lib/trie/database.go

```diff
@@ -316,12 +347,14 @@ func (t *Trie) GetChangedNodeHashes() (inserted, deleted map[string]struct{}, er
 	inserted = make(map[string]struct{})
 	err = t.getInsertedNodeHashesAtNode(t.root, inserted)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, fmt.Errorf("getting inserted node hashes: %w", err)
 	}
 
-	deleted = make(map[string]struct{}, len(t.deletedMerkleValues))
-	for k := range t.deletedMerkleValues {
-		deleted[k] = struct{}{}
+	deletedNodeHashes := t.deltas.Deleted()
+	// TODO return deletedNodeHashes directly after changing MerkleValue -> NodeHash
```

Comment:
done in draft subsequent PR #2915

Reply:
maybe link this in comment unless will be merged into this branch?

Reply:
Eh it's just 2 PRs away (#2917 is based on this one, #2915 is based on #2917), let's keep it as is. It won't be merged in this PR (for the glorious git blame), but it should be merged to development rather soon.